src/transformers/models/mask2former/configuration_mask2former.py

```diff
@@ -0,0 +1,334 @@
+        mask_feature_size=256,
+        hidden_dim=256,
+        encoder_feedforward_dim=1024,
+        norm="GN",
+        encoder_layers=6,
+        decoder_layers=10,
+        num_heads=8,
+        dropout=0.1,
+        dim_feedforward=2048,
+        pre_norm=False,
+        enforce_input_proj=False,
+        common_stride=4,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        self.feature_size = feature_size
+        self.mask_feature_size = mask_feature_size
+        self.hidden_dim = hidden_dim
+        self.encoder_feedforward_dim = encoder_feedforward_dim
+        self.norm = norm
+        self.encoder_layers = encoder_layers
+        self.decoder_layers = decoder_layers
+        self.num_heads = num_heads
+        self.dropout = dropout
+        self.dim_feedforward = dim_feedforward
+        self.pre_norm = pre_norm
+        self.enforce_input_proj = enforce_input_proj
+        self.common_stride = common_stride
+
+    @classmethod
+    def from_pretrained(cls, pretrained_model_name_or_path: Union[str, os.PathLike], **kwargs) -> "PretrainedConfig":
+
+        config_dict, kwargs = cls.get_config_dict(pretrained_model_name_or_path, **kwargs)
+
+        # get the decoder config dict if we are loading from Mask2FormerConfig
+        if config_dict.get("model_type") == "mask2former":
+            config_dict = config_dict["decoder_config"]
+
+        if "model_type" in config_dict and hasattr(cls, "model_type") and config_dict["model_type"] != cls.model_type:
+            logger.warning(
+                f"You are using a model of type {config_dict['model_type']} to instantiate a model of type "
+                f"{cls.model_type}. This is not supported for all configurations of models and can yield errors."
+            )
+
+        return cls.from_dict(config_dict, **kwargs)
+
+
+class Mask2FormerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`Mask2FormerModel`]. It is used to instantiate a
+    Mask2Former model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the Mask2Former
+    [facebook/mask2former-instance-swin-small-coco](facebook/mask2former-instance-swin-small-coco) architecture trained
+    on [ADE20k-150](https://huggingface.co/datasets/scene_parse_150).
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Currently, Mask2Former only supports the [Swin Transformer](swin) as backbone.
+
+    Args:
+        backbone_config (`PretrainedConfig` or `dict`, *optional*, defaults to `SwinConfig()`):
+            The configuration of the backbone model. if unset, the configuration corresponding to
+            `swin-base-patch4-window12-384` will be used.
+        decoder_config (`dict`, *optional*):
```

Comment:
Maybe for consistency with the backbone config, this could be both a `PretrainedConfig` or `dict`?